include/SDL3/SDL_events.h

```diff
@@ -615,7 +620,27 @@ typedef struct SDL_GamepadButtonEvent
 /**
  * Gamepad device event structure (event.gdevice.*)
  *
+ * Joysticks that are supported Gamepads receive both an SDL_JoyDeviceEvent and
+ * an SDL_GamepadDeviceEvent.
+ *
+ * SDL will populate the event queue with GAMEPAD_ADDED events for joysticks
+ * that are already plugged in duing SDL_Init and are recognized as Gamepads.
+ *
+ * GAMEPAD_ADDED events for controllers plugged in at the start of the program
+ * will not occur automatically if their mappings are added after calling
+ * SDL_Init.
+ *
+ * If you rely on GAMEPAD_ADDED events to tell you about controllers that are
+ * already plugged in at the start of the program, and you use
+ * SDL_AddGamepadMappingsFromFile to add support for additional controllers,
+ * you can ensure that you get GAMEPAD_ADDED events for all controllers if you
+ * load the mappings before calling SDL_Init.
+ *
  * \since This struct is available since SDL 3.0.0.
+ *
+ * \sa SDL_AddGamepadMappingsFromFile
+ * \sa SDL_HINT_GAMECONTROLLERCONFIG_FILE
```

Comment:
It looks like it's still called GAMECONTROLLERCONFIG_FILE in SDL 3. Should it be renamed to GAMEPADCONFIG_FILE?

Reply:
No, that's intentional for compatibility between SDL2 and SDL3.

I'll double check, I think you're supposed to get events for gamepads when their mappings become available. If not, we can probably change that.

Reply:
I just noticed SDL 3 has [SDL_ReloadGamepadMappings](https://wiki.libsdl.org/SDL3/SDL_ReloadGamepadMappings), so maybe that's a new feature in SDL 3? My testing has been with SDL 2, so I'm not sure.